wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitService.java

```diff
@@ -90,76 +100,88 @@
+            gitConnection.add(params);
         }
     }
 
-    @Path("checkout")
     @POST
+    @Path("checkout")
     @Consumes(MediaType.APPLICATION_JSON)
     public void checkout(CheckoutRequest request) throws ApiException {
         try (GitConnection gitConnection = getGitConnection()) {
-            gitConnection.checkout(request);
+            CheckoutParams params = CheckoutParams.create(request.getName())
+                                                  .withFiles(request.getFiles())
+                                                  .withCreateNew(request.isCreateNew())
+                                                  .withNoTrack(request.isNoTrack())
+                                                  .withTrackBranch(request.getTrackBranch())
+                                                  .withStartPoint(request.getStartPoint());
+            gitConnection.checkout(params);
         }
     }
 
-    @Path("branch-create")
     @POST
+    @Path("branch")
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces(MediaType.APPLICATION_JSON)
     public Branch branchCreate(BranchCreateRequest request) throws ApiException {
         try (GitConnection gitConnection = getGitConnection()) {
-            return gitConnection.branchCreate(request);
+            return gitConnection.branchCreate(request.getName(), request.getStartPoint());
         }
     }
 
-    @Path("branch-delete")
-    @POST
-    @Consumes(MediaType.APPLICATION_JSON)
-    public void branchDelete(BranchDeleteRequest request) throws ApiException {
+    @DELETE
+    @Path("branch")
+    public void branchDelete(@QueryParam("name") String name, @QueryParam("force") boolean force) throws ApiException {
         try (GitConnection gitConnection = getGitConnection()) {
-            gitConnection.branchDelete(request);
+            gitConnection.branchDelete(name, force);
         }
     }
 
-    @Path("branch-rename")
     @POST
+    @Path("branch")
     public void branchRename(@QueryParam("oldName") String oldName,
                              @QueryParam("newName") String newName) throws ApiException {
         try (GitConnection gitConnection = getGitConnection()) {
             gitConnection.branchRename(oldName, newName);
         }
     }
 
-    @Path("branch-list")
-    @POST
-    @Consumes(MediaType.APPLICATION_JSON)
-    @Produces({MediaType.APPLICATION_JSON, MediaType.TEXT_PLAIN})
-    public GenericEntity<List<Branch>> branchList(BranchListRequest request) throws ApiException {
+    @GET
+    @Path("branch")
+    @Produces({MediaType.APPLICATION_JSON})
+    public GenericEntity<List<Branch>> branchList(@QueryParam("listMode") String listMode) throws ApiException {
         try (GitConnection gitConnection = getGitConnection()) {
-            return new GenericEntity<List<Branch>>(gitConnection.branchList(request)) {
+            return new GenericEntity<List<Branch>>(gitConnection.branchList(listMode == null ? null : BranchListMode.valueOf(listMode))) {
```

Comment:
BranchListMode.valueOf(listMode) can throw IllegalArgumentException. Mb we should wrap it to BadRequestException?